internal/tool/prepare.go

```diff
@@ -0,0 +1,134 @@
+package tool
+
+import (
+	"bufio"
+	"os"
+	"path"
+	"strings"
+
+	codacy "github.com/codacy/codacy-engine-golang-seed/v6"
+	"github.com/samber/lo"
+)
+
+const sourceConfigFileName = ".semgrep.yaml"
+
+// TODO: should respect cli flag for docs location
+const rulesDefinitionFileName = "/docs/rules.yaml"
+
+func createConfigFile(toolExecution codacy.ToolExecution) (*os.File, error) {
+
+	if toolExecution.Patterns == nil {
+		return createConfigFileFromScratch(toolExecution)
+	}
+
+	if len(*toolExecution.Patterns) == 0 {
+		return nil, nil
+	}
+
+	// if there are patterns, create a configuration file from them
+	return createConfigFileFromPatterns(toolExecution.Patterns)
+}
+
+func createConfigFileFromScratch(toolExecution codacy.ToolExecution) (*os.File, error) {
+	// if there is no configuration file use default configuration file
+	// otherwise use configuration from source code
```

Comment:
It this correct? Shouldn't it be:

```suggestion
	// Use the tool's configuration file, if it exists.
	// Otherwise use the tool's default patterns.
```

Reply:
Amidst the refactor I didn't re-read that, thanks!